protocols/imap/src/main/java/org/apache/james/imap/processor/GetAnnotationProcessor.java

```diff
@@ -0,0 +1,139 @@
+
+            final long maxsize = message.getMaxsizeValue();
+            if (maxsize > 0) {
+                ImmutableSortedSet<Integer> maxsizeValues = getValuesOverSize(mailboxAnnotations, maxsize);
+
+                if (!maxsizeValues.isEmpty()) {
+                    responder.respond(new AnnotationResponse(mailboxName, filterItemsBySize(responder, mailboxName, mailboxAnnotations, maxsize)));
+                    okComplete(command, tag, StatusResponse.ResponseCode.longestMetadataEntry(maxsizeValues.first()), responder);
+                    return;
+                }
+            }
+            responder.respond(new AnnotationResponse(mailboxName, mailboxAnnotations));
+            okComplete(command, tag, responder);
+        } catch (MailboxNotFoundException e) {
+            session.getLog().info(command.getName() + " failed for mailbox " + mailboxName, e);
+            no(command, tag, responder, HumanReadableText.FAILURE_NO_SUCH_MAILBOX, StatusResponse.ResponseCode.tryCreate());
+        } catch (MailboxException e) {
+            session.getLog().info(command.getName() +" failed for mailbox " + mailboxName, e);
+            no(command, tag, responder, HumanReadableText.GENERIC_FAILURE_DURING_PROCESSING);
+        }
+    }
+
+    private List<MailboxAnnotation> filterItemsBySize(Responder responder, String mailboxName, List<MailboxAnnotation> mailboxAnnotations, final long maxsize) {
+        Predicate<MailboxAnnotation> lowerPredicate = new Predicate<MailboxAnnotation>() {
+            public boolean apply(MailboxAnnotation input) {
+                return (input.getValue().get().getBytes().length < maxsize);
+            }
+        };
+
+        return FluentIterable.from(mailboxAnnotations).filter(lowerPredicate).toList();
+    }
+
+    private List<MailboxAnnotation> getMailboxAnnotations(ImapSession session, Set<String> keys, GetAnnotationRequest.Depth depth, MailboxPath mailboxPath) throws MailboxException {
+        switch (depth) {
+            case ZERO:
+                if (keys.isEmpty()) {
+                    return getMailboxManager().getAllAnnotations(mailboxPath, ImapSessionUtils.getMailboxSession(session));
+                } else {
+                    return getMailboxManager().getAnnotationsByKeys(mailboxPath, ImapSessionUtils.getMailboxSession(session), keys);
+                }
+
+            case ONE:
+                return getMailboxManager().getAnnotationsByKeysWithOneDepth(mailboxPath, ImapSessionUtils.getMailboxSession(session), keys);
+
+            case INFINITY:
+                return getMailboxManager().getAnnotationsByKeysWithAllDepth(mailboxPath, ImapSessionUtils.getMailboxSession(session), keys);
+
+            default:
+                return ImmutableList.of();
+        }
+
+    }
+
+    private ImmutableSortedSet<Integer> getValuesOverSize(List<MailboxAnnotation> mailboxAnnotations, final long maxsize) {
```

Comment:
See my other comment, here you just need to return the max value, not the entire collection.